docs/conventions/component.md

```diff
@@ -0,0 +1,461 @@
+
+In high-level way of thinking, the base should serve as "an interface" and systems' components as specific implementations. Inheritance, though, should be achieved by composition and not by polymorphism.
+
+That's why the base should also be thought of, and managed as, a common layer of re-usable set of rules and guidelines.
+
+All components should follow the same file structure and naming conventions.
+
+If you are not familiar with the concept of systems in ECL, first have a look at [the system documentation](../systems).
+
+### Naming of components
+
+* Throughout the whole project, in `package.json` files, namespace should always be `@ecl/`, regardless whether work is to be done on a generic or system component. One reason is that `@ecl` namespace should be preserved for correct organization on [npmjs](https://www.npmjs.com/).
+* Flavored components should be prefixed with the name of their corresponding system and type (component, style, template): `[generic|ec|eu]-[component|style|template|utility]`.
+
+Example:
+```
+// Some full component name (in package.json)
+@ecl/ec-component-logo
+@ecl/generic-component-navigation-menu
+@ecl/eu-utility-flex
+
+// Corresponding component name (used for file name later)
+ec-component-logo
+generic-component-navigation-menu
+eu-utility-flex
+```
+
+## SCSS
+
+Apart from [rules that should be applied to all scss files](scss), there are some specifications for components:
+
+### File name
+
+SCSS file name should be the same as component's name.
+
+Example:
+```
+generic-component-navigation-menu.scss
+```
+
+### Generic component
+
+All css rules should be put in mixins.
+
+It could be split in several small mixins, but in any cases we should have a main mixin calling the small ones.
+This main mixin should be named following the component's main class.
+
+Example:
+```scss
+@mixin ecl-navigation-menu-common() {
+  // Some css rules or mixin include
+}
+@mixin ecl-navigation-menu-responsive() {
+  // Some css rules
+}
+@mixin ecl-navigation-menu() {
+  @include ecl-navigation-menu-common();
+  @include ecl-navigation-menu-responsive();
+}
+```
+
+The css rules in generic components should not define specific display elements (like color, background, and more globally everything specific to a system).
+
+If display is different in several systems, these differences should be put as parameters in the mixin. The goal here is to be able to call the same generic code, and just pass some parameters to alter display. There is no need to repeat component's name for these parameters, as they are limited to this mixin.
+
+Example:
+```scss
+@mixin ecl-navigation-menu(
+  $bar-bg-mobile,
+  $bar-bg-desktop,
+  $toggle-btn-color,
+  $group-bg,
+  $separator-mobile,
+  $link-color,
+  $active-color,
+  $active-bg
+) {
+  // Some css rules or mixin include
+}
+```
+
+### Systems component
+
+SCSS files in system-specific components rely on generic SCSS file.
```

Comment:
`generic SCSS files` as it's a general context, or `SCSS files from generic base are dependencies to system-specific components` to state it closer in terms of dependency management.